### PR TITLE
LaTeX export

### PR DESCRIPTION
**Pull request by olaure01**
OCaml part of the LaTeX export with missing ingredients:
* [x] does not turn ending numbers to sub-script for output
* [ ] explicit exchange rule only so far, implicit one could be considered as a proposed option

The front-end part remains to be done.

### CODE REVIEW

**Comment by etiennecallies** [comment]
Deployed on https://linearon.modusponens.dev/?s=A*A%5E%2CA%7CA%5E+

**Comment by olaure01** [comment]
Technical error for open proofs.

**Comment by etiennecallies** [comment]
> Technical error for open proofs.

Indeed, fixed and redeployed

File: export_as_latex.ml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+open Proof
+
+exception Cannot_export_proof_as_latex_exception of string;;
+
+let proof_to_latex proof =
+    let header = "% This Latex file has been generated using Click&coLLect tool.\n"
```
Comment (olaure01) [inline export_as_latex.ml:6]:
```suggestion
    let header = "% This LaTeX file has been generated using Click&coLLect tool.\n"
```

File: latex/example.tex [hunk 2]
```diff
@@ -1,4 +1,12 @@
-\documentclass{article}
+% This Coq file has been generated using Click&coLLect tool.
```
Comment (olaure01) [inline latex/example.tex:1]:
```suggestion
% This LaTeX file has been generated using Click&coLLect tool.
```

File: export_as_latex.ml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        ^ "\\newcommand*{\\withv}[1]{\\infer{2}[\\ensuremath{\\with}]{\\vdash #1}}\n"
+        ^ "\\newcommand*{\\pluslv}[1]{\\infer{1}[\\ensuremath{\\plus_1}]{\\vdash #1}}\n"
+        ^ "\\newcommand*{\\plusrv}[1]{\\infer{1}[\\ensuremath{\\plus_2}]{\\vdash #1}}\n"
+        ^ "\\newcommand*{\\ocv}[1]{\\infer{1}[\\ensuremath{\\oc}]{\\vdash #1}}\n"
+        ^ "\\newcommand*{\\wkv}[1]{\\infer{1}[\\ensuremath{?\\mathit{w}}]{\\vdash #1}}\n"
+        ^ "\\newcommand*{\\cov}[1]{\\infer{1}[\\ensuremath{?\\mathit{c}}]{\\vdash #1}}\n"
+        ^ "\\newcommand*{\\dev}[1]{\\infer{1}[\\ensuremath{?\\mathit{d}}]{\\vdash #1}}\n\n" in
+    let start_proof = "\\begin{document}\n\n\\begin{prooftree*}\n" in
+    let proof_lines = Proof.to_latex proof in
+    let end_proof = "\\end{prooftree*}\n\n\\end{document}\n\n" in
```
Comment (olaure01) [inline export_as_latex.ml:38]:
```suggestion
    let end_proof = "\\end{prooftree}\n\n\\end{document}\n\n" in
```

File: export_as_latex.ml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        ^ "\\newcommand*{\\tensorv}[1]{\\infer{2}[\\ensuremath{\\tensor}]{\\vdash #1}}\n"
+        ^ "\\newcommand*{\\parrv}[1]{\\infer{1}[\\ensuremath{\\parr}]{\\vdash #1}}\n"
+        ^ "\\newcommand*{\\withv}[1]{\\infer{2}[\\ensuremath{\\with}]{\\vdash #1}}\n"
+        ^ "\\newcommand*{\\pluslv}[1]{\\infer{1}[\\ensuremath{\\plus_1}]{\\vdash #1}}\n"
+        ^ "\\newcommand*{\\plusrv}[1]{\\infer{1}[\\ensuremath{\\plus_2}]{\\vdash #1}}\n"
+        ^ "\\newcommand*{\\ocv}[1]{\\infer{1}[\\ensuremath{\\oc}]{\\vdash #1}}\n"
+        ^ "\\newcommand*{\\wkv}[1]{\\infer{1}[\\ensuremath{?\\mathit{w}}]{\\vdash #1}}\n"
+        ^ "\\newcommand*{\\cov}[1]{\\infer{1}[\\ensuremath{?\\mathit{c}}]{\\vdash #1}}\n"
+        ^ "\\newcommand*{\\dev}[1]{\\infer{1}[\\ensuremath{?\\mathit{d}}]{\\vdash #1}}\n\n" in
+    let start_proof = "\\begin{document}\n\n\\begin{prooftree*}\n" in
```
Comment (olaure01) [inline export_as_latex.ml:36]:
```suggestion
    let start_proof = "\\begin{document}\n\n\\begin{prooftree}\n" in
```

File: latex/example.tex [hunk 5]
```diff
@@ -7,42 +15,44 @@
...
+\newcommand*{\pluslv}[1]{\infer{1}[\ensuremath{\plus_1}]{\vdash #1}}
+\newcommand*{\plusrv}[1]{\infer{1}[\ensuremath{\plus_2}]{\vdash #1}}
+\newcommand*{\ocv}[1]{\infer{1}[\ensuremath{\oc}]{\vdash #1}}
+\newcommand*{\wkv}[1]{\infer{1}[\ensuremath{?\mathit{w}}]{\vdash #1}}
+\newcommand*{\cov}[1]{\infer{1}[\ensuremath{?\mathit{c}}]{\vdash #1}}
+\newcommand*{\dev}[1]{\infer{1}[\ensuremath{?\mathit{d}}]{\vdash #1}}
 
 \begin{document}
 
 \begin{prooftree*}
```
Comment (olaure01) [inline latex/example.tex:39]:
```suggestion
\begin{prooftree}
```

File: latex/example.tex [hunk 6]
```diff
@@ -7,42 +15,44 @@
...
+  \exv{\bot\plus\bot, \top}
+  \axv{A, A\orth}
+  \tensorv{\bot\plus\bot, \top\tensor A, A\orth}
+  \exv{\bot\plus\bot, A\orth, \top\tensor A}
+  \parrv{(\bot\plus\bot)\parr A\orth, \top\tensor A}
+  \dev{\wn ((\bot\plus\bot)\parr A\orth), \top\tensor A}
+  \ocv{\wn ((\bot\plus\bot)\parr A\orth), \oc (\top\tensor A)}
+  \wkv{\wn ((\bot\plus\bot)\parr A\orth), \wn ((\bot\plus\bot)\parr A\orth), \oc (\top\tensor A)}
+  \cov{\wn ((\bot\plus\bot)\parr A\orth), \oc (\top\tensor A)}
 \end{prooftree*}
```
Comment (olaure01) [inline latex/example.tex:56]:
```suggestion
\end{prooftree}
```

**Review by olaure01 — COMMENTED** [review]
Better rendering: better crop.

File: sequent.ml [hunk 7]
```diff
@@ -145,9 +145,68 @@ let formula_list_to_coq formula_list =
...
+let rec formula_to_latex_atomic =
+  function
+  | One -> "\\one", true
+  | Bottom -> "\\bot", true
+  | Top -> "\\top", true
+  | Zero -> "\\zero", true
+  | Litt x -> litteral_to_latex x, true
+  | Orth e ->
+      let s, atomic = formula_to_latex_atomic e in
+      let s_parenthesis = if atomic then s else "(" ^ s ^ ")" in
```
Comment (olaure01) [inline sequent.ml:166]:
```suggestion
      let s_parenthesis = if atomic then "{" ^ s ^ "}" else "(" ^ s ^ ")" in
```

**Comment by etiennecallies** [comment]
Thanks a lot @olaure01 ! Let's merge this PR, and develop implicit exchange rule in another PR.

**Pull request merged**